cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/WorkflowDataset.java

```diff
@@ -82,6 +82,25 @@ void write(Id.Workflow id, RunRecordMeta runRecordMeta, List<ProgramRun> program
     table.put(rowKey, NODES, Bytes.toBytes(value));
   }
 
+  public void delete(Id.Application id) {
+    MDSKey mdsKey = new MDSKey.Builder().add(id.getNamespaceId()).add(id.getId()).build();
+    Scanner scanner = table.scan(mdsKey.getKey(), Bytes.stopKeyForPrefix(mdsKey.getKey()));
+    Row row;
+    try {
+      while ((row = scanner.next()) != null) {
+        MDSKey.Splitter splittedKey  = new MDSKey(row.getRow()).split();
+        String namespaceId = splittedKey.getString();
+        splittedKey.skipString();
+        String appId = splittedKey.getString();
+        if (namespaceId.equals(id.getNamespaceId()) && appId.equals(id.getId())) {
```

Comment:
I don't think this comparison is needed, since you're building the row key with the namespace and appId